src/job/job.service.ts

```diff
@@ -56,6 +56,28 @@ export class JobService {
     };
   }
 
+  async getJobsByEvent(eventId: string) {
+    const jobs = await this.prismaService.job.findMany({
+      where: { eventId },
+      include: {
+        executions: {
+          orderBy: { createdAt: 'desc' },
+          take: 1,
+          include: { transaction: true },
+        },
+      },
+    });
+
+    return jobs.map((job) => ({
+      ...job,
+      executions: job.executions.map((execution) => ({
```

Comment:
I am not sure about API design but I want to suggest that naming as `latestExecution` since it takes only one execution. Or Is it for expandability? **It is not strong opinion.**

Reply:
Thank you for your suggestion. I spread executions field into single depth.